### PR TITLE
SPMORANGLE-48 Staff—manage my own tasks (RBAC: staff)

### PR DESCRIPTION
**Pull request by Qingyu255**
This pull request introduces collaborator-aware access control for tasks, allowing the system to determine and expose whether a user has edit permissions on a task (either as owner or collaborator). The changes span service interfaces, implementations, repositories, DTOs, controller, and tests to support this feature.

**Collaborator access control and exposure:**

* Added `isRequestingUserTaskCollaborator(Long taskId)` to `UserContextService` and its implementation, enabling checks if the current user is a collaborator for a given task. [[1]](diffhunk://#diff-2717010a2d0d60c231cad59acd1078c359bf7a69f8081b09e91e2c5c5496a97bR17-R18) [[2]](diffhunk://#diff-3695002f82e8baa2ccdf373ca13073558c154dc5845735146f1f01499f4bbb62R18) [[3]](diffhunk://#diff-3695002f82e8baa2ccdf373ca13073558c154dc5845735146f1f01499f4bbb62R34-R38)
* Extended `CollaboratorService` and its implementation to support checking collaborator status and retrieving all tasks a user collaborates on. [[1]](diffhunk://#diff-97ced1bce2766f262fc837fc578a724aeeb467c3e4a47e75b74c2d57b7238f6bR7-R18) [[2]](diffhunk://#diff-182f4e941f0b65213f3ca559cffd607d42ef7e3d892464d42d6697d573a44c2dR60-R69)
* Updated `TaskAssigneeRepository` with methods and queries for collaborator checks and task retrieval by user.

**Task service and DTO changes:**

* Modified `TaskService` and its implementation so task queries (e.g., `getProjectTasks`) are collaborator-aware and return a new `userHasEditAccess` field in `TaskResponseDto`. This field is set based on ownership or collaboration status. [[1]](diffhunk://#diff-1da68f1d24540572b18860295190e33b17c3f46001ec4073d70a98e99ddcc3e4L13-R22) [[2]](diffhunk://#diff-02061d064fe7cf84e08d44b7a2144125ca0174e2819f9fa42d069704c5c4a988L87-R106) [[3]](diffhunk://#diff-02061d064fe7cf84e08d44b7a2144125ca0174e2819f9fa42d069704c5c4a988L109-R115) [[4]](diffhunk://#diff-02061d064fe7cf84e08d44b7a2144125ca0174e2819f9fa42d069704c5c4a988L134-R151) [[5]](diffhunk://#diff-02061d064fe7cf84e08d44b7a2144125ca0174e2819f9fa42d069704c5c4a988R164) [[6]](diffhunk://#diff-9f110a1338b99a51887df7f15dab3d0143405dc9535fb0b492bc4171d183aa42R25)

**Controller update:**

* Changed `TaskManagementController` to pass the requesting user's ID to task queries, enabling collaborator-aware filtering and access information.

**Test coverage:**

* Added comprehensive unit tests for the new collaborator check in `UserContextServiceTest`, covering positive, negative, and edge cases. [[1]](diffhunk://#diff-06af7a67b21cd47c9480d38bbfbc9def94019200ad03268f91ed500e8ad9889dR7) [[2]](diffhunk://#diff-06af7a67b21cd47c9480d38bbfbc9def94019200ad03268f91ed500e8ad9889dR32-R34) [[3]](diffhunk://#diff-06af7a67b21cd47c9480d38bbfbc9def94019200ad03268f91ed500e8ad9889dR223-R407)

### CODE REVIEW

File: backend/spmorangle/src/main/java/com/spmorangle/crm/taskmanagement/service/TaskService.java [hunk 1]
```diff
@@ -10,12 +10,14 @@ public interface TaskService {
...
-    List<TaskResponseDto> getProjectTasks(Long projectId);
+    List<TaskResponseDto> getProjectTasks(Long userId, Long projectId);
     
     List<TaskResponseDto> getPersonalTasks(Long userId);
     
     List<TaskResponseDto> getAllUserTasks(Long userId);
     
     void deleteTask(Long taskId, Long currentUserId);
 
+    boolean canUserUpdateOrDeleteTask(Long taskId, Long userId);
```
Comment (Qingyu255) [inline backend/spmorangle/src/main/java/com/spmorangle/crm/taskmanagement/service/TaskService.java:21]: @jitthing boss can use this method to do RBAC check on your task update/delete endpoints

File: frontend/components/task-card.tsx [hunk 2]
```diff
@@ -223,15 +224,28 @@ export function TaskCard({ task, variant = 'board', onSubtaskUpdated }: TaskCard
           <Dialog open={showDetails} onOpenChange={setShowDetails}>
             <DialogTrigger asChild>
               <Button variant="ghost" size="sm">
-                View Details
+                {taskProps.userHasEditAccess ? (
+                  <>Manage</>
```
Comment (Qingyu255) [inline frontend/components/task-card.tsx:228]: @jitthing i leave this to you when youre working on the delete and update functionality?

File: frontend/components/task-card.tsx [hunk 3]
```diff
@@ -223,15 +224,28 @@ export function TaskCard({ task, variant = 'board', onSubtaskUpdated }: TaskCard
...
+              <DialogHeader className="flex flex-row justify-between">
+                <div>
+                  <DialogTitle>{taskProps.title}</DialogTitle>
+                  <DialogDescription>
+                    {taskProps.key} • {taskProps.project}
+                  </DialogDescription>
+                </div>
+                {taskProps.userHasEditAccess && (
+                  <div className="px-2 flex flex-row space-x-2">
+                    {/* @jitt Please replace with actual update and delete functionality */}
```
Comment (Qingyu255) [inline frontend/components/task-card.tsx:244]: here too
